dashboard/src/lib/string.ts

```diff
@@ -8,3 +8,26 @@ export const truncateBigText = (
   text && text.length > maxTextLength
     ? text.slice(0, maxTextLength) + '...'
     : valueOrEmpty(text);
+
+export const truncateUrl = (
+  url: string | undefined,
+  domainLength = 50,
+  endPathLength = 20,
+): string => {
+  if (url) {
+    if (url.length <= domainLength + endPathLength) {
+      return url;
+    }
+    const protocolRe = /^\w+:\/\//;
+    const searchParamsRe = /\?.*$/;
+    let replacedUrl = url.replace(protocolRe, '');
+    replacedUrl = replacedUrl.replace(searchParamsRe, '');
+    const splittedUrl = replacedUrl.split('/');
+    const hostname = splittedUrl[0];
```

Comment:
```suggestion
    const hostname = splittedUrl[0] ?? '';
```

typescript strict mode doesn't check array index

Reply:
I'm checking for undefined below